README.md

```diff
@@ -2,40 +2,75 @@
-model to represent its accounts. This means that smart contracts
-written for the Cardano ecosystem are different from the usual
-account-based ledger format we see in most other blockchain ecosystems.
+[systemf]: https://en.wikipedia.org/wiki/System_F
+[incorrectness]: https://dl.acm.org/doi/pdf/10.1145/3371078
+[tweag-blogpost]: TODO
+
+# Use Cases
+
+## Example Language
+
+The `Language.Pirouette.Example` defines the `Ex` language: pirouette's example language.
+We refer the interested reader to check the respective module for guidelines on using pirouette over
+your own language.
 
-The core of a EUTxO contract is the _[validator script](https://plutus.readthedocs.io/en/latest/plutus/tutorials/basic-validators.html)_ and in a multitude of scenarios, these validator scripts will be created with the help
-of either the _StateMachine_ library [[1](https://plutus-pioneer-program.readthedocs.io/en/latest/week7.html#code-example-2),[2](https://github.com/input-output-hk/plutus/blob/master/plutus-contract/src/Plutus/Contract/StateMachine.hs)] or a custom implementation of state machines. Therefore, the core of most contracts will be a Mealy Machine of the form:
+Now, say we have the simple program:
 
 ```haskell
-transition :: State -> Input -> Maybe (State, Output)
+addone :: Integer -> Integer
+addone x = x + 1
 ```
 
-Pirouette's main functionality is extracting the `transition` function into TLA+ in such a way that
-the constructors of the `Input` datatype are used to represent the possible state transitions. This means
-that when TLA+ finds a counter-example, we will see the sequence `[i0, ..., iN] :: [Input]` of inputs
-we need to pass to `transition` to witness the failure.
+And we wish to check the incorrectness triple `[x > 0] addone x [result > 0]`, that
+is: if `addone x > 0`, then `x > 0`. We obviously expect a counterexample with `x = 0`.
+We can use pirouette to find that assignment:
+
+```haskell
+params :: IncorrectnessParams Ex
+params = IncorrectnessParams
+          [term| \x:Integer . x + 1 |] -- A function term to symbolically evaluate
+          [ty| Integer |] -- type of the result of the function term above
+          -- Conditions to check, they take the result and the inputs of the function term 
+          ([term| \(res:Integer) (x:Integer) . 0 < res |]
+            :==>: [term| \(res:Integer) (x:Integer) . 0 < x |])
+```
+
+Now we can run `replIncorrectnessLogic1 10 params` in a repl and we should see:
+```
+💸 COUNTEREXAMPLE FOUND
+{ __result ↦ 1
+  x ↦ 0 }
+```
+
+## Plutus Smart Contracts
+
+[Plutus] is a subset of Haskell used to
+write smart contracts for Cardano blockchain, which utilizes the _Extended UTxO_
+[[1](https://iohk.io/en/research/library/papers/the-extended-utxo-model/),
+[2](https://iohk.io/en/research/library/papers/native-custom-tokens-in-the-extended-utxo-model/)]
+model to represent its accounts. Pirouette has a prototype interface to interacting with
```

Comment:
I'm not sure the EUTxO part is worth mentioning here in the context of Pirouette.

The folks who know about Plutus likely know this, and the folks who don't probably don't care.